models/dataset.go

```diff
@@ -454,24 +454,43 @@ func (ed *EditionUpdate) PublishLinks(ctx context.Context, host string, versionL
 
 // ValidateDataset checks the dataset has invalid fields
 func ValidateDataset(ctx context.Context, dataset *Dataset) error {
+	var generalDetails = []*GeneralDetails{}
```

Comment:
You are creating this empty array and passing that into the `trimHref` func. It's empty so isn't doing anything and its not linked to the `dataset` var so it won't update the fields.

I think what you meant to do is pass in one of the `dataset` fields.